mitodl/open-discussions
# Added VideoResource indexing

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2307 

#### What's this PR do?
- Adds `VideoResource` as an indexed object type
- Adds task helper methods for #2305 
- Refactored identical implementations of `index_profiles`, `index_bootcamps`, etc into a common `index_items`

#### How should this be manually tested?
Create one or more `VideoResource`s:

```python
from course_catalog.factories import VideoResourceFactory
from search.task_helpers import upsert_video
vr1 = VideoResourceFactory.create()
upsert_video(vr1)
vr2 = VideoResourceFactory.create()
```

Ensure the only first one shows up in the search page and facets work as expected.

Then you can reindex: `./manage.py recreate_index` and ensure both videos show up now.
